src/shims/os_str.rs

```diff
@@ -237,7 +237,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_ref();
         let os_str = this.read_os_str_from_wide_str(scalar)?;
 
-        Ok(PathBuf::from(&convert_path_separator(&os_str, &this.tcx.sess.target.target.target_os, Pathconversion::TargetToHost)))
+        Ok(PathBuf::from(&convert_path_separator(Cow::Borrowed(&os_str), &this.tcx.sess.target.target.target_os, Pathconversion::TargetToHost)))
```

Comment:
No this doesn't help. :) The nice part is that you can now do `Cow::Owned(os_str)`. That is what avoids the copy.
Also I think it's better to use `.into_owned()` instead of `PathBuf::from`; makes it more explicit that this is a `Cow` operation.

Reply:
Thank you for the correction! 😺